### PR TITLE
Add post-mortem september 2023

### PR DESCRIPTION
**Pull request by saraycp**

### CODE REVIEW

File: _posts/deployments/2023-09-19-post-mortem.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
+---
+layout: post
+title: "Post-mortem: Gap Between Deployment and Migration End"
+category: deployments
+---
+
+## OBS Pages Inaccessible on 19th of September
+
+After the morning deployment, many of the OBS pages were inaccessible returning a 500 error for less than 13 minutes.
+The error messages had to do with `moderated_at` and `code_of_conduct` database columns that the deployment tried to introduce.
```
Comment (hennevogel) [inline _posts/deployments/2023-09-19-post-mortem.md:10]: None of your readers have the possibility to see that...

File: _posts/deployments/2023-09-19-post-mortem.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+**Resolution**: Everything went back to normal when the migrations finished. No human intervention.
+
+**Detection**: Manual check after deployment and error tracking tool.
+
+## Lessons Learned
+
+**What went well?**
+
+* We started the incident protocol as soon as possible.
+* Another team member assumed the communicator role quickly.
```
Comment (danidoni) [inline _posts/deployments/2023-09-19-post-mortem.md:29]: I think this point may be important for us, but not for the people outside the team. I would drop it.

File: _posts/deployments/2023-09-19-post-mortem.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+**What went wrong?**
+
+* During development, [strong_migrations](https://github.com/ankane/strong_migrations) didn't detect any unsafe migration on our code.
+* We weren't able to foresee that the migration could take longer than expected.
```
Comment (hennevogel) [inline _posts/deployments/2023-09-19-post-mortem.md:34]:
```suggestion
* We did not test how long the migration runs.
```

**Review by hennevogel — CHANGES_REQUESTED** [review]

File: _posts/deployments/2023-09-19-post-mortem.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
+---
+layout: post
+title: "Post-mortem: Gap Between Deployment and Migration End"
+category: deployments
+---
+
+## OBS Pages Inaccessible on 19th of September
+
+After the morning deployment, many of the OBS pages were inaccessible returning a 500 error for less than 13 minutes.
```
[the remark this answers is not in the document]
Comment (saraycp) [reply _posts/deployments/2023-09-19-post-mortem.md]: Not only, on the same timeframe we also got [these erros](https://errbit.opensuse.org/apps/5620ca2bdc71fa00b1000000/problems/65095b8e84bf460703acd7be) about the `code_of_conduct` column.

[the remark this answers is not in the document]
Comment (saraycp) [reply _posts/deployments/2023-09-19-post-mortem.md]: Well, the errors with `conde_of_conduct` happened in less than one minute, not so relevant... 
`First Notice: Sep 19 08:27:58 Last Notice: Sep 19 08:28:31`

**Comment by saraycp** [comment]
@hennevogel I already push all the changes you suggested. Despite the fact

**Review by danidoni — APPROVED** [review]

**Review by eduardoj — APPROVED** [review]

**Comment by ncounter** [comment]
@saraycp what is the state of this PR?

**Comment by saraycp** [comment]
> @saraycp what is the state of this PR?

The blog post missed the Action Items section. I needed time to do some research and testing in order to provide possible solutions. As it is not straightforward, I've decided to add a generic text and go on working on it after this post-mortem is published:

_"Plan a strategy to deploy migrations. It might include separating migration deployment from the code changes deployment or splitting the migration into different steps"_

**Comment by bmwiedemann** [comment]
> "Plan a strategy to deploy migrations. It might include separating migration deployment from the code changes deployment or splitting the migration into different steps"

That is a sensible approach. First we do the required changes to the database, ensure everything keeps working as before and in a 2nd deployment we add code to use the new DB parts. Would that split happen on the git level, where the PR to use the new parts would wait longer for merge?

**Comment by saraycp** [comment]
> > "Plan a strategy to deploy migrations. It might include separating migration deployment from the code changes deployment or splitting the migration into different steps"
> 
> That is a sensible approach. First we do the required changes to the database, ensure everything keeps working as before and in a 2nd deployment we add code to use the new DB parts. Would that split happen on the git level, where the PR to use the new parts would wait longer for merge?

The order _migration deployment_ + _changes deployment_ works for the addition of columns/tables.
For removals, the order should be  _changes deployment_ + _migration deployment_, as we need the code to stop using those columns/tables before we remove them.

So far, we have relied on labels like "Reference Server" or "Do Not Merge" to avoid merging and deploying PRs that require special attention. We can use them for the cases we mentioned before until we find a better way to do that.

**Review by hennevogel — APPROVED** [review]

File: _posts/deployments/2023-10-09-post-mortem.md [hunk 5]
```diff
@@ -0,0 +1,50 @@
+---
+layout: post
+title: "Post-mortem: Service Degradation for Pages with Comments"
+category: deployments
+---
+
+## OBS Pages Inaccessible on 19th of September
+
+On September 19th the pages and API routes on OBS displaying comments where inaccessible (returning a 500 error) to all users for 13 minutes. Here is what caused the problem.
```
Comment (dmarcoux) [inline _posts/deployments/2023-10-09-post-mortem.md:9]:
```suggestion
On September 19th the pages and API routes on OBS displaying comments were inaccessible (returning a 500 error) to all users for 13 minutes. Here is what caused the problem.
```

**Pull request merged**